lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -675,11 +675,12 @@ public function matching(Criteria $criteria): Collection
     /**
      * Retrieves the wrapped Collection instance.
      *
-     * @return Collection<TKey, T>
+     * @return Collection<TKey, T>&Selectable<TKey, T>
      */
     public function unwrap(): Collection
```

Comment:
In the matching method, there is the code
```
public function matching(Criteria $criteria): Collection
    {
        if ($this->isDirty) {
            $this->initialize();
        }

        if ($this->initialized) {
            return $this->unwrap()->matching($criteria);
        }
```
So, unwrap need to return a Selectable if we want to call `matching` on it.